pkg/download/dependency_resolution/resolver/shared.go

```diff
@@ -80,24 +81,36 @@ func replaceAll(content string, key string, s string) string {
 
 		log.Debug("Failed to replace %q with %q in string values in %q: %s", key, s, content, err)
 	}
-	return replaceAllUsingRegEx(content, key, s)
+	return replaceAllUsingRegEx(content, key, s, configType)
 }
 
-func replaceAllUsingRegEx(content string, key string, s string) string {
-	// The prefix and suffix we search for are alphanumerical, as well as the "-", and "_".
-	// From investigating, this character set seems to be the most basic regex that still avoids false positive substring matches.
-	str := fmt.Sprintf("([^a-zA-Z0-9_-])(%s)([^a-zA-Z0-9_-])", key)
-
-	// replace only strings that are not part of another larger string. See testcases for exact in/out values.
-	re, err := regexp.Compile(str)
+func replaceAllUsingRegEx(content string, key string, s string, configType config.Type) string {
+	re, err := getMatchingRegEx(key, configType)
 	if err != nil {
-		log.Debug("Failed to compile string %q to regex. Falling back to use simple string replace.", str)
+		slog.Debug("Failed to compile regex. Falling back to use simple string replace.", slog.String("key", key), slog.String("configType", string(configType.ID())))
 		return strings.ReplaceAll(content, key, s)
 	}
 
 	return re.ReplaceAllString(content, fmt.Sprintf("$1%s$3", s))
 }
 
+// getMatchingRegEx returns a regex for finding the specified key related to the config type.
+func getMatchingRegEx(key string, configType config.Type) (*regexp.Regexp, error) {
+	if featureflags.RestrictDocumentReferenceCreation.Enabled() {
+		switch configType.(type) {
```

Comment:
Can't that just be `configType.ID() == types.DocumentTypeID`?

Reply:
Done 👍